src/agent/coverage/src/cobertura.rs

```diff
@@ -1,12 +1,56 @@
 use crate::source::SourceCoverage;
-use crate::source::SourceCoverageLocation;
 use crate::source::SourceFileCoverage;
 use anyhow::Context;
 use anyhow::Error;
 use anyhow::Result;
+use std::path::Path;
 use std::time::{SystemTime, UNIX_EPOCH};
 use xml::writer::{EmitterConfig, XmlEvent};
 
+pub fn compute_line_rate(line_values: Vec<u32>) -> f32 {
```

Comment:
Let's use this function to work through a few Rust things.

1. This function will panic if `line_values` has less than 2 elements. We never want to panic in our code, so we should only call functions that can panic if we're sure that the panic case is unreachable for all inputs. In this case, the (notation-hidden) function is [`Index::index()`](https://doc.rust-lang.org/std/ops/trait.Index.html#tymethod.index). However, this gets fixed if we address (2) below.
2. These indices actually have a _fixed_ meaning (count of lines hit, valid). When that's true, a function like this should either take 2 arguments, or take 1 argument that is a meaning-encoding struct. For example, something like `struct LineCounts { hit: u32, count: u32 }`. In this case, I think 2 args is probably more simple and straightforward.
3. We typically don't want to take ownership over a non-`Copy` value unless we need to. If we kept the existing signature, we'd probably want this argument to have type `line_values: &[u32]`. We'd want a `&[u32]` instead of a `&Vec<u32>` because the only `Vec` methods we're using actually come from `Deref`-ing through to slices anyway. It almost never makes sense to pass a `&Vec<T>`, because the most `Vec`-specific methods require a _mutable_ reference to the `Vec`. An example of an exception: when you might want to clone the `Vec` itself inside the function.
4. Since we only support 64-bit, our machine word size is 64 bits, and there's no semantic use in limiting these computations to `u32`/`f32`. I suggest thinking of the "default" numeric types for our purposes as `u64`/`i64`/`f64`. Especially when counting things in Rust, the "usual" numeric type is `usize` (which is also used for indexing), but here I'd suggest an explicit `u64`. If the Cobertura format only supports up to 32-bit numbers, then your current choice is the right one, but we should document it!

Reply:
Makes, sense, fixed for this function and all 32-bit types changed to 64.

Reply:
Removed this function and replaced with line values type that computes line rate, as suggested in other comment.